ports/mimxrt/hal/startup_MIMXRT10xx.S

```diff
@@ -0,0 +1,166 @@
+
+    .syntax unified
+    .arch armv7-m
+
+#include"flexram_config.s"
+
+
+/* Reset Handler */
+
+    .thumb_func
+    .align 2
+    .globl   Reset_Handler
+    .type    Reset_Handler, %function
+Reset_Handler:
+    cpsid   i               /* Mask interrupts */
+    .equ    VTOR, 0xE000ED08
+    ldr     r0, =VTOR
+    ldr     r1, =__isr_vector
+    str     r1, [r0]
+    ldr     r2, [r1]
+    msr     msp, r2
+
+/* Reconfigure the memory map, which must match the setting of the linker script */
+    dsb
+    isb
+    ldr r0, =__iomux_gpr17_adr   /* load IOMUXC_GPR17 register address to R0 */
+    ldr r1, =__iomux_gpr17_value /* move FlexRAM configuration value to R1 */
+    str r1,[r0]                  /* store FLEXRAM configuration value to IOMUXC_GPR17 */
+    dsb
+    isb
+    ldr r0, =__iomux_gpr16_adr  /* load IOMUXC_GPR16 register address to R0 */
+    ldr r1,[r0]                 /* load IOMUXC_GPR16 register value to R1 */
+    orr r1, r1, #4              /* set corresponding FLEXRAM_BANK_CFG_SEL bit */
+    str r1,[r0]                 /* store the value to IOMUXC_GPR16 (FLEXRAM_BANK_CFG_SEL = '1') */
+    dsb
+    isb
+
+#ifndef __NO_SYSTEM_INIT
+    ldr   r0,=SystemInit
+    blx   r0
+#endif
+/*     Loop to copy data from read only memory to RAM. The ranges
+ *      of copy from/to are specified by following symbols evaluated in
+ *      linker script.
+ *      __etext: End of code section, i.e., begin of data sections to copy from.
+ *      __data_start__/__data_end__: RAM address range that data should be
+ *      __noncachedata_start__/__noncachedata_end__ : none cachable region
+ *      __ram_function_start__/__ram_function_end__ : ramfunction region
+ *      copied to. Both must be aligned to 4 bytes boundary.  */
+
+    ldr    r1, =__etext
+    ldr    r2, =__data_start__
+    ldr    r3, =__data_end__
+
+#ifdef __PERFORMANCE_IMPLEMENTATION
```

Comment:
The difference in code size between `__PERFORMANCE_IMPLEMENTATION` enabled and disabled is very small, and these chips have a huge amount of flash, so I think it's simpler/cleaner to just unconditionally use the performance version of the code (ie remove the other code).

Reply:
Hi, I tried it out now but it does not work. Even when just compiling with `__PERFORMANCE_IMPLEMENTATION` defined, MicroPython crashes.
If you are ok with it, let us merge it as it is and we will look into it later and fix it.